cli: CLI prints out previous error message when executing a quit command

hanpen24 opened the pull request:
Previously, when executing the quit command in the CLI, the last SQL error message would be displayed again. 
This was inadequate because there is no need to show a previous SQL error when executing a quit command. 
To address this, this patch modifies the behavior so that when commands like quit or \q are executed, any previous error messages are cleared.

Fixes: #85613

Release note (bug fix): Fixed a bug where the CLI would print out
the previous error message when executing the quit command.

hanpen24 committed:
cli: CLI prints out previous error message when executing a quit command

Previously, when executing the quit command in the CLI, the last
SQL error message would be displayed again. This was inadequate
because there is no need to show a previous SQL error when
executing a quit command. To address this, this patch modifies
the behavior so that when commands like quit or \q are executed,
any previous error messages are cleared.

Fixes: #85613

Release note (bug fix): Fixed a bug where the CLI would print out
the previous error message when executing the quit command.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130736)
<!-- Reviewable:end -->

rafiss reviewed (approved):
thanks for your contribution!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130736#-:-O6ktat--m2V-JGd5R2n:bvdps3i)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
it seems that this change caused an existing test to fail, so this may not be the right solution.

https://github.com/cockroachdb/cockroach/blob/5d9f6cb7f4a0a272af26c6c560d8debc7c2b8395/pkg/cli/interactive_tests/test_last_statement.tcl#L14-L23

The test expects to see `hello 1` in the output (with `1` being the exit code from the CLI), but it doesn't see that.

hanpen24 commented:
Thank you for pointing that out. I'll check it.

hanpen24 commented:
@rafiss 
I apologize for the delayed response.

In this test case, we expect that even when the shell is explicitly exited with \q, the exit status of 1 from the previous command will be displayed. However, if the user intentionally exits the shell with \q, it should be considered a normal exit, and the exit status should be 0.

Therefore, I believe the expected result in the test case should be revised to "hello 0."

rafiss commented:
@hanpen24 Thank you for investigating the test. I agree with your reasoning - so let's adjust the test to expect a 0 exit code (and change any related comments in the test).

hanpen24 commented:
@rafiss 
Thank you for your feedback! I have updated the test to expect a 0 exit code as discussed, and I have also modified the relevant comments in the test. Please let me know if there are any further adjustments needed.

rafiss reviewed (approved):
thank you for your contribution!

bors r+
